### PR TITLE
Deepfried dd Slide Edits

### PR DESCRIPTION
**Pull request by RMenezes782**
Reorderings and enhancements:
1. Moved block definition to the start of the presentation, right after introduction in 'real dd's explanation.
2. Placed real dd's usage before applications for a more intuitive ordering.
3. Added more applications as 'advanced applications'.
4. Separated all warnings into their own slide and infinite size files and infinitely running dd.
5. Separated 'our dd's implementation slides and added mention of signal handling.
6. Revamped old 'blocks' slide to instead have an example use case of our dd.
7. Added our common 'Questions?' slide at the end.

### CODE REVIEW

File: _slides/deepfried_dd.md [hunk 1]
```diff
@@ -5,22 +5,21 @@ title: Deepfried_DD
...
 
-## Applications of `dd`
+## What is a Block?
 
-* Data transfer, and in-place modification of files
-* Wiping disks for security (e.g. prior to recycling hardware)
-  - Contents of a file are not necessarily overwritten on regular deletion
-* Generating files with random data (will be useful for testing in upcoming Networking MP)
-* "Flashing" / installing custom ROMs (enhanced versions of vanilla OS) on Android devices
+* A unit measuring the number of bytes that are read or written at one time.
```
Comment (breidys2) [inline _slides/deepfried_dd.md:13]: This definition of block is a bit imprecise. I would just call it “a sequence of bits with a predefined size ( the block size)”

File: _slides/deepfried_dd.md [hunk 2]
```diff
@@ -5,22 +5,21 @@ title: Deepfried_DD
...
-## Applications of `dd`
+## What is a Block?
 
-* Data transfer, and in-place modification of files
-* Wiping disks for security (e.g. prior to recycling hardware)
-  - Contents of a file are not necessarily overwritten on regular deletion
-* Generating files with random data (will be useful for testing in upcoming Networking MP)
-* "Flashing" / installing custom ROMs (enhanced versions of vanilla OS) on Android devices
+* A unit measuring the number of bytes that are read or written at one time. 
+* Hard drives / SSDs have a sector (block) size of 4 kB
```
Comment (breidys2) [inline _slides/deepfried_dd.md:14]: *Typically* have a block size of 4kb

File: _slides/deepfried_dd.md [hunk 3]
```diff
@@ -30,7 +29,42 @@ title: Deepfried_DD
 
 <horizontal/>
 
-## What our dd will look like
+## Applications of `dd`
```
Comment (breidys2) [inline _slides/deepfried_dd.md:32]: Tbh, would move this before the command line parameters

Comment (RMenezes782) [reply _slides/deepfried_dd.md:32]: I'd say here the problem is that students can't really engage meaningfully in a discussion of application until they know what the utility _does_. I feel that the params give a good description of dd usage and possibilities before we abstract the conversation to uses.

Comment (breidys2) [reply _slides/deepfried_dd.md:32]: Fair enough

**Review by breidys2 — APPROVED** [review]
LGTM

**Pull request merged**